### ISSUE TITLE
Button Outline: Default wrong padding

### ISSUE THREAD

**Issue by hanneslsm** [WordPress/gutenberg#52636]
### Description

I have two buttons.
One is the default, with a border-top 2px and border-bottom 2px.
The second is the outline version, also 2px defined as border.
I'd expect both have the same height, but this seems not to be the case, because some default padding is applied. I could manually define the padding for both buttons so that they have both the same height, but shouldn't the default values be the same?

### Step-by-step reproduction instructions

1. Create a new blank theme
2. Define the border for the default border, e.g. top & bottom 2px
3. Optionally, add 2px border to the outline style

### Screenshots, screen recording, code snippet

https://github.com/WordPress/gutenberg/assets/17861443/839506ff-d811-40b2-99df-8b65e6f8caaf

### Environment info

- WP 6.2
- GB 16.2.1
- Custom theme

### Please confirm that you have searched existing issues in the repo.

No

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by felipevelzani** [comment]
Hi there,

Thank you for your report!

Do you mean that when you change the border on the Default button, the border/dimensions aren't reflected for Outline?

It is interesting that I see it as expected on the editor, but I see a similar issue on the site itself. Do you reproduce the same way?

<img width="1494" alt="Screenshot 2023-07-17 at 18 12 30" src="https://github.com/WordPress/gutenberg/assets/40555737/a28493cc-ee0b-4764-8924-f48968af7e7a">

**Comment by hanneslsm** [comment]
> Do you mean that when you change the border on the Default button, the border/dimensions aren't reflected for Outline?

No. What I mean is that the height differs from the default and outline variant, even though (from my understanding) there shouldn't be a difference.

https://github.com/WordPress/gutenberg/assets/17861443/d673c1d2-dba8-402f-9f23-f61543effddd


> It is interesting that I see it as expected on the editor, but I see a similar issue on the site itself. Do you reproduce the same way?

Sorry, I didn't understand?

**Comment by t-hamano** [comment]
My understanding is that this is the correct behavior. 

Applying the Outline block style will draw a **2px** border. The button has `box-sizing: border-box` applied, but it has no explicit height, so the block with outline style height is **4px** larger. [4px is explicitly subtracted](https://github.com/WordPress/gutenberg/blob/49174f2479529d6a493e7ce46f27b1f818850fd3/packages/block-library/src/button/style.scss#L33-L34) to make this height match the default height.

However, in the Border panel, values other than 2px can be set, so it is not possible to decide how many px the height should be subtracted.

**Comment by hanneslsm** [comment]
> [4px is explicitly subtracted](https://github.com/WordPress/gutenberg/blob/49174f2479529d6a493e7ce46f27b1f818850fd3/packages/block-library/src/button/style.scss#L33-L34) to make this height match the default height.
.
However, in the Border panel, values other than 2px can be set, so it is not possible to decide how many px the height should be subtracted.

This seems to be the issue. 2px shouldn't be a hard coded value.

**Comment by hanneslsm** [comment]
@foosantos are there any more info needed for this issue as the label indicates?

**Comment by felipevelzani** [comment]
Hi, @hanneslsm! I apologize for the delay in following up here.

As @t-hamano said, this is expected, but I totally agree that it is an odd approach. I believe it is something that we definitely should consider reviewing.

I honestly can't see a clear path here, as it would probably not be user-friendly to show a default negative padding for the outline when users open the options (which explains this being hardcoded). 

I even wondered about removing 2px from the padding for the Default/Fill button when a custom border is added to it, but it wouldn't fix 100% of this issue and would add a lot more complexity to it.

**Comment by t-hamano** [comment]
The underlying problem was probably that the stretch option wasn't working properly. That issue was resolved in https://github.com/WordPress/gutenberg/pull/64770.

Now, all buttons should be the same height, regardless of the thickness of their outline:

![Image](https://github.com/user-attachments/assets/21f3913b-2a38-4069-a45a-77fdad39d859)

I would like to close this issue, but feel free to re-open if I've missed something.
